apps/site/assets/js/test/support-form_test.js

```diff
@@ -571,7 +572,56 @@ describe("support form", () => {
       event.initEvent("change", false, false);
       sortBySelect.dispatchEvent(event);
 
-      assert.isFalse($("#routeAndVehicle").is(":visible"));
+      // $("#...").is(":visible") doesn't detect visibility so we need to use the 'display' CSS property instead:
+      assert.equal($("#routeAndVehicle").css("display"), "none");
+    });
+  });
+
+  describe("handleSubjectChange", () => {
+    beforeEach(() => {
+      $("#test").html(`
+       <select class="c-select c-mode-selector" id="support_subject">
+         <option value>Please choose a subject</option>
+         <option value="Bus%20Stop">Bus Stop</option>
+         <option value="CharlieCards%20%26%20Tickets">CharlieCards &amp; Tickets</option>
+       </select>
+       <div id="charlie-card-or-ticket-number" class="form-group">
+        <label class="form-control-label" for="support_ticket_number">CharlieCard or Ticket number (optional)</label>
+        <input class="support-form-text-input support-form-text-input--small form-control" id="support_ticket_number" name="support[ticket_number]" type="text">
+       </div>
+     `);
+      handleSubjectChange($);
+    });
+
+    afterEach(() => {
+      $("#test").remove();
+    });
+
+    it("shows the 'CharlieCard or Ticket number' field", () => {
```

Comment:
For the sake of someone scanning through the tests, it might be handy to include something in these test descriptions about in what circumstances it does or does not show the field.